src/azure-cli-core/azure/cli/core/util.py

```diff
@@ -22,11 +22,25 @@
 def handle_exception(ex):
     # For error code, follow guidelines at https://docs.python.org/2/library/sys.html#sys.exit,
     from msrestazure.azure_exceptions import CloudError
+    from msrest.exceptions import HttpOperationError
     if isinstance(ex, (CLIError, CloudError)):
         logger.error(ex.args[0])
         return ex.args[1] if len(ex.args) >= 2 else 1
     elif isinstance(ex, KeyboardInterrupt):
         return 1
+    elif isinstance(ex, HttpOperationError):
+        try:
+            response_dict = json.loads(ex.response.json())
+            if 'error' in response_dict:
```

Comment:
If you want to be super secure, having "error" in response_dict does not tell you that there "code" or "message" in "error". You could either catch KeyError (if error is there, but not "message" for instance), or replace with something like:
```python
response_dict["error"].get('code', 'Unknown code')
```
(I would suggest the last one).

Reply:
So according to [odata v4](http://docs.oasis-open.org/odata/odata-json-format/v4.01/cs01/odata-json-format-v4.01-cs01.html#_Toc499720637):
This object MUST contain name/value pairs with the names code and message, and it MAY contain name/value pairs with the names target, details and innererror.

So I assumed that if ARM follows odata v4 and if there is an `error` key the key's value must be a dict with a `code` and `message` key.  

But if indeed I shouldn't take anything for granted with ARM, I will update the code to be more secure.

Thanks for the heads up @lmazuel .

Reply:
I will also add some unittests.

Reply:
@adewaleo The API review is stronger than it was at some point, so you're *likely* right that error will follow odata v4. But CLI should assume ARM is full of traps :)